src/main/java/io/supertokens/Main.java

```diff
@@ -275,7 +275,7 @@ private void createDotStartedFileForThisProcess() throws IOException {
         boolean ignored = dotStarted.setWritable(true, false);
         this.startedFileName = fileName;
         try (BufferedWriter writer = new BufferedWriter(new FileWriter(dotStarted))) { // overwrite mode
-            writer.write(ProcessHandle.current().pid() + "");
+            writer.write(ProcessHandle.current().pid() + "\n" + Config.getConfig(this).getBasePath());
```

Comment:
- We should consider normalising the value returned by `getBasePath` ([here](https://github.com/supertokens/supertokens-core/blob/master/src/main/java/io/supertokens/config/CoreConfig.java#L95)):
   - Make sure it has a `/` prefix if it is not an empty string
   - Make sure it has no spaces (if it does, we should throw a `QuitProgramException` [from this function](https://github.com/supertokens/supertokens-core/blob/master/src/main/java/io/supertokens/config/CoreConfig.java#L205)
   - Remove any trailing `/`s

We should write tests for the above normalisation / validation as well in [this file](https://github.com/supertokens/supertokens-core/blob/master/src/test/java/io/supertokens/test/ConfigTest2_6.java).

- It would be good to add tests [in this file](https://github.com/supertokens/supertokens-core/blob/master/src/test/java/io/supertokens/test/DotStartedFileTest.java):
   - Check that by default, the file indicates an empty base path
   - If we set a base path, the .started file should have that

Reply:
The `validateAndInitialize` method [checks the path for spaces](https://github.com/supertokens/supertokens-core/blob/75c20bc032d2915bf1492036351d6a0160c3d499/src/main/java/io/supertokens/config/CoreConfig.java#L277) and throws a `QuitProgramException` if it does. However, I have also added this logic to the `getBasePath` method. I have also added the rest of the base path normalization logic that you requested.